docs/src/main/asciidoc/class-loading-reference.adoc

```diff
@@ -0,0 +1,23 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - Class Loading Reference
```

Comment:
+100 for this doc!

It would be wonderful to have this document completed so people that haven't followed the ClassLoader system design discussions can have a good reference point and perhaps even be able to contribute somewhat to the review process